### PR TITLE
HPCC-32439 Make all warnings into errors in Roxie subdirectory

### PR DESCRIPTION
**Pull request by richardkchapman**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [x] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by richardkchapman** [comment]
This MAY require changes in your branch to compile - I think you made some fixes in roxiemem. I can take out the change to add strict to that dir if preferred

File: roxie/roxiemem/CMakeLists.txt [hunk 1]
```diff
@@ -22,6 +22,7 @@
 #####################################################
 
 project( roxiemem ) 
+SET (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${STRICT_CXX_FLAGS}")
```
Comment (richardkchapman) [inline roxie/roxiemem/CMakeLists.txt:25]: This may cause issues until your change is merged

File: roxie/udplib/udptrr.cpp [hunk 2]
```diff
@@ -1538,7 +1536,7 @@ class CReceiveManager : implements IReceiveManager, public CInterface
         //any other sender for the udpPermitTimeout period
         unsigned maxSlotsPerClient = (udpMaxPendingPermits == 1) ? queue_size : (udpMaxClientPercent * queue_size) / (udpMaxPendingPermits * 100);
         assertex(maxSlotsPerClient != 0);
-        if (maxSlotsPerClient > queue_size)
+        if ((int) maxSlotsPerClient > queue_size)
```
Comment (richardkchapman) [inline roxie/udplib/udptrr.cpp:1539]: Could argue that changing type of queue_size would be preferable but I didn't want to go down that rabbit hole

File: roxie/udplib/udptrr.cpp [hunk 3]
```diff
@@ -953,8 +953,6 @@ class CReceiveManager : implements IReceiveManager, public CInterface
         void noteDone(UdpSenderEntry *requester, const UdpRequestToSendMsg &msg)
         {
             const unsigned flowSeq = msg.flowSeq;
-            SendPermit * permit = requester->queryPermit(flowSeq);
```
Comment (richardkchapman) [inline roxie/udplib/udptrr.cpp:956]: This was a change I stared at for a bit to try to convince myself that the warning was not hinting at buggy code. I _think_ it's just an unnecessary call.

**Comment by ghalliday** [comment]
It looks like some other changes (constructor order) need merging first.

**Review by ghalliday — COMMENTED** [review]
Looks reasonable

**Review by ghalliday — APPROVED** [review]
Looks reasonable

**Comment by ghalliday** [comment]
Various issues from failing smoketests

**Comment by ghalliday** [comment]
From windows build:
D:\a\HPCC-Platform\HPCC-Platform\HPCC-Platform\roxie\udplib\udpsim.cpp(312,117): error C3493: 'maxSendQueueSize' cannot be implicitly captured because no default capture mode has been specified [D:\a\HPCC-Platform\HPCC-Platform\build\roxie\udplib\udpsim.vcxproj]

From mac:
/Users/runner/work/HPCC-Platform/HPCC-Platform/HPCC-Platform/roxie/udplib/udpmsgpk.cpp:222:30: error: variable 'packetDataSize' set but not used [-Werror,-Wunused-but-set-variable]
                    unsigned packetDataSize = fingerHdr->length - fingerHdr->metalength - sizeof(UdpPacketHeader);
                             ^
/Users/runner/work/HPCC-Platform/HPCC-Platform/HPCC-Platform/roxie/udplib/udpmsgpk.cpp:71:14: error: private field 'numPackets' is not used [-Werror,-Wunused-private-field]
    unsigned numPackets = 0;
             ^
2 errors generated.

**Comment by ghalliday** [comment]
I think it may still be failing to build on windows - closed and reopened to check

**Comment by ghalliday** [comment]
@richardkchapman still failing on windows

**Review by ghalliday — CHANGES_REQUESTED** [review]
Still failing on windows

File: roxie/udplib/udptrr.cpp (hunk 3 above)
Comment (ghalliday) [reply roxie/udplib/udptrr.cpp:956]: yes, I agree

File: roxie/roxiemem/roxiemem.cpp [hunk 4]
```diff
@@ -8425,8 +8425,6 @@ class RoxieMemTests : public CppUnit::TestFixture
 
         for (unsigned pass=0; pass < 8; pass++)
         {
-            //numPagesAfterCleanup() also has the side-effect of freeing up any empty pages.
```
Comment (ghalliday) [inline roxie/roxiemem/roxiemem.cpp:8428]: This looks like it should be retained.  It may have a side-effect used in the test.

**Review by ghalliday — COMMENTED** [review]
@richardkchapman one line and comment that I think should stay in (already fixed on a previous PR)

**Review by ghalliday — APPROVED** [review]
@richardkchapman please squash and I will merge

**Comment by richardkchapman** [comment]
Squashed

**Pull request merged**